Subject: disable PREEMPT_SPINLOCK_WAITERS when x86 ticket/fifo spins are in use

From: Gregory Haskins <ghaskins@novell.com>

Preemptible spinlock waiters effectively bypasses the benefits of a fifo
spinlock.  Since we now have fifo spinlocks for x86 enabled, disable the
preemption feature on x86.

Signed-off-by: Gregory Haskins <ghaskins@novell.com>
CC: Nick Piggin <npiggin@suse.de>
---

 arch/x86/Kconfig |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
index 8d15667..d5b9a67 100644
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -20,6 +20,7 @@ config X86
 	bool
 	default y
 	select HAVE_MCOUNT
+	select DISABLE_PREEMPT_SPINLOCK_WAITERS
 
 config GENERIC_TIME
 	bool
